odl/util/utility.py

```diff
@@ -707,6 +709,44 @@ def signature_string(posargs, optargs, sep=', ', mod=''):
     return part_sep.join(parts)
 
 
+def unique(seq):
+    """Return the unique values in a sequence.
+
+    Parameters
+    ----------
+    seq : sequence
+        Sequence with (possibly duplicate) elements.
+
+    Returns
+    -------
+    unique : list
+        Unique elements of seq. Order is guaranteed to be the same as in seq.
+
+    Examples
+    --------
+    >>> unique([1, 2, 3, 3])
+    [1, 2, 3]
+
+    >>> unique((1, 'str', 'str'))
+    [1, 'str']
+
+    Also works with unhashable types:
+
+    >>> unique((1, [1], [1]))
+    [1, [1]]
+    """
+    # First check if all elements are hashable, if so O(n) can be done
+    try:
+        return list(OrderedDict.fromkeys(seq))
```

Comment:
It seems not:
https://stackoverflow.com/questions/1653970/does-python-have-an-ordered-set